lib/fs-hash-diff.js

```diff
@@ -0,0 +1,77 @@
+/// @ts-check
+"use strict";
+
+const Entry = require("fs-tree-diff/lib/entry").default;
+const FSTree = require("fs-tree-diff");
+
+class HashEntry {
+  /**
+   * @param relativePath {string}
+   * @param hash {string}
+   */
+  constructor(relativePath, hash) {
+    this.relativePath = relativePath;
+    this.hash = hash;
+  }
+
+  isDirectory() {
+    return false;
+  }
+
+  /**
+   * Whether the entries have the same content.
+   * @param other {Entry | HashEntry}
+   * @returns
+   */
+  equals(other) {
+    if (other instanceof Entry) {
```

Comment:
I think an exact class check is wrong here. this is an instance method on HashEntry so any subclasses of this can use this implementation or override it. I can just invert this test to check if it's an instance of `HashEntry` instead of the other way around.